src/main/java/org/sagebionetworks/bridge/services/AccountService.java

```diff
@@ -286,6 +289,34 @@ public void deleteAccount(AccountId accountId) {
         }
     }
     
+    /**
+     * Delete all accounts that are preview users in this study. We check and throw an exception 
+     * if someone attempts to enroll a preview user in more than one study, so it's safe to 
+     * just delete here.
+     */
+    public void deleteAllPreviewAccounts(String appId, String studyId) {
+        checkNotNull(appId);
+        checkNotNull(studyId);
+        
+        Set<String> allOfGroups = ImmutableSet.of(PREVIEW_USER_GROUP);
+        AccountSummarySearch.Builder searchBuilder = new AccountSummarySearch.Builder()
+                .withPageSize(API_MAXIMUM_PAGE_SIZE)
+                .withAllOfGroups(allOfGroups)
+                .withEnrolledInStudyId(studyId);
+        PagedResourceList<AccountSummary> page = null;
+        int offset = 0;
+        do {
+            searchBuilder.withOffsetBy(offset);
+            page = getPagedAccountSummaries(appId, searchBuilder.build());
+            for (AccountSummary summary : page.getItems()) {
+                // It is too slow to use deleteAccount because it cleans up a ton of
+                // DynamoDB resources. So... we leave all the non-relational data behind.
+                accountDao.deleteAccount(summary.getId());
+            }
+            offset += API_MAXIMUM_PAGE_SIZE;
+        } while(offset < page.getTotal());
```

Comment:
I'm not sure about this, but is it possible to miss preview accounts if there are more to delete than the API_MAXIMUM_PAGE_SIZE? For instance, if the max page size is 100 and there are 125 preview accounts, it would delete 100 and set the offset to 100. Then the next time through the getPagedAccountSummaries() call would have an offset of 100, bypassing the remaining 25 records since the first 100 were already deleted.

Reply:
GOOD CATCH. All we have to do here is keep calling until the list comes back empty, from offset 0 each time. I had initially been collecting all the IDs, then switched late in the writing of the code!